mav_trajectory_generation_ros/src/ros_visualization.cpp

```diff
@@ -200,4 +200,34 @@ void drawVertices(const Vertex::Vector& vertices, const std::string& frame_id,
                                 marker_array);
 }
 
+void drawVerticesFromTrajectory(const Trajectory& trajectory,
+                                const std::string& frame_id,
+                                visualization_msgs::MarkerArray* marker_array) {
+  CHECK_NOTNULL(marker_array);
+
+  std::vector<double> segment_times = trajectory.getSegmentTimes();
+  if (segment_times.empty()) {
+    ROS_WARN("Empty trajectory.");
+    return;
+  }
+  Vertex::Vector vertices(segment_times.size() + 1, trajectory.D());
+
+  double trajectory_time = 0.0;
+  for (size_t i = 0; i < vertices.size(); ++i) {
+    vertices[i] =
+        trajectory.getVertexAtTime(trajectory_time, derivative_order::POSITION);
+    if (i < segment_times.size() - 1) trajectory_time += segment_times[i];
```

Comment:
that line is a bit hard to understand - the minus 1 - maybe could be integrated with the following assertion somehow, to make it clear why trajectory_time is not update further.